routes/socket.js

```diff
@@ -1,43 +1,42 @@
 const SocketServer = require('socket.io');
 const redisAdapter = require('socket.io-redis');
+const gameRoomService = require('../service/gameRoomService');
 const { initLogger } = require('../utils/logger');
 
 const logger = initLogger(module);
 
-const rooms = {
-  // eslint-disable-next-line quote-props
-  '223455': 'e92e95f8-71cc-455d-9426-07744ea2b83d',
-  // eslint-disable-next-line quote-props
-  '102345': 'c4bad8e3-3145-461a-9d28-1e9cacfb856b',
-  // eslint-disable-next-line quote-props
-  '987931': 'e29cc8f0-16c3-4c23-83ec-41224d77bf8d'
-};
-
-const getRoomId = (joinPin) => rooms[joinPin] || null;
+const getRoomId = (joinPin) => Promise.resolve(gameRoomService.findFromPin(joinPin))
+  .then((value) => value.id);
 
 const handleSocketConnection = (io, socket) => {
   const { token } = socket.handshake.query;
   logger.info('Client connecting...', { token });
 
-  const roomId = getRoomId(token);
+  const roomId = Promise.resolve(getRoomId(token)).then((room) => {
```

Comment:
Og viss du bruker async/await for eksempel her, så slipper du å måtte nøste alt inn i .then callbacks

Reply:
Mener du i getRoomId? for det funker ikke å sette handlesocketconnection til async